src/main/scala/com/typesafe/sbt/bundle/SbtBundle.scala

```diff
@@ -175,6 +201,24 @@ object SbtBundle extends AutoPlugin {
     hashArchive
   }
 
+  private def createConfiguration(): Def.Initialize[Task[File]] = Def.task {
+    val bundleTarget = (target in BundleConfiguration).value
+    val configurationTarget = (NativePackagerKeys.stage in BundleConfiguration).value
+    def relParent(p: (File, String)): (File, String) =
+      (p._1, configurationName.value + java.io.File.separator + p._2)
+    val configChildren: List[File] = configurationTarget.listFiles().toList
+    val bundleMappings: Seq[(File, String)] = configChildren.flatMap(_.pair(relativeTo(configurationTarget)))
+    val archive = Archives.makeZip(bundleTarget, configurationName.value, bundleMappings, None)
+    val archiveName = archive.getName
+    val exti = archiveName.lastIndexOf('.')
+    val hash = Hash.toHex(digestFile(archive))
+    val hashName = archiveName.take(exti) + "-" + hash + archiveName.drop(exti)
+    val hashArchive = archive.getParentFile / hashName
+    IO.move(archive, hashArchive)
+    streams.value.log.info(s"Configuration Bundle has been created: $hashArchive")
+    hashArchive
+  }
```

Comment:
A lot of the above is doing what `shazar` does in case you're looking for inspiration when it comes to names.